mlir/include/mlir/Target/LLVMIR/LLVMImportInterface.h

```diff
@@ -155,7 +165,19 @@ class LLVMImportInterface
   LogicalResult convertIntrinsic(OpBuilder &builder, llvm::CallInst *inst,
                                  LLVM::ModuleImport &moduleImport) const {
     // Lookup the dialect interface for the given intrinsic.
-    Dialect *dialect = intrinsicToDialect.lookup(inst->getIntrinsicID());
+    llvm::Intrinsic::ID intrinId = inst->getIntrinsicID();
```

Comment:
```suggestion
    // Verify the intrinsic identifier maps to an actual intrinsic.
    llvm::Intrinsic::ID intrinId = inst->getIntrinsicID();
```
nit: The comment above should probably move downs the the lookup?

I wonder if we can ever get not_intrinsic?

Reply:
> I wonder if we can ever get not_intrinsic?

I'm trying to defend against `llvm::CallInst *inst`, but the call site for `convertIntrinsic` current covers this with `dyn_cast<llvm::IntrinsicInst>(inst)` - I'll turn this into an assert instead.